src/js/notifications.js

```diff
@@ -0,0 +1,190 @@
+        return markUnregistered()
+        .then(() => body);
+      }
+      return body;
+    });
+  }
+  return shouldUnregister()
+  .then(reloadUI);
+}
+
+function register(feature) {
+  navigator.serviceWorker.ready
+  .then(registration => registration.pushManager.getSubscription()
+    .then(subscription => {
+      if (subscription) {
+        // already subscribed to a(nother?) feature
+        return subscription;
+      }
+
+      return registration.pushManager.subscribe({ userVisibleOnly: true });
+    }))
+  .then(subscription => {
+    const rawKey = subscription.getKey ? subscription.getKey('p256dh') : '';
+    const key = rawKey ?
+              btoa(String.fromCharCode.apply(null, new Uint8Array(rawKey))) :
+              '';
+    const rawAuthSecret = subscription.getKey ? subscription.getKey('auth') : '';
+    const authSecret = rawAuthSecret ?
+              btoa(String.fromCharCode.apply(null, new Uint8Array(rawAuthSecret))) :
+              '';
+
+    return fetch('/register', {
+      method: 'post',
+      headers: { 'Content-type': 'application/json' },
+      body: JSON.stringify({
+        deviceId,
+        endpoint: subscription.endpoint,
+        key,
+        authSecret,
+        features: [feature],
+      }),
+    });
+  })
+  .then(handleRegistrationsResponse);
+}
+
+function unregister(feature) {
+  return fetch('/unregister', {
+    method: 'post',
+    headers: { 'Content-type': 'application/json' },
+    body: JSON.stringify({
+      deviceId,
+      features: (feature) ? [feature] : null,
+    }),
+  })
+  .then(handleRegistrationsResponse);
+}
+
+function toggleNotification(event) {
+  const notificationElement = event.target;
+  const feature = notificationElement.dataset.slug;
+  // sometimes a click to `undefined` happened
+  if (feature) {
+    if (notificationElement.dataset.notification === 'true') {
+      console.log('DEBUG: Unregistering from', feature);
+      return unregister(feature);
+    }
+    console.log('DEBUG: Registering to', feature);
+    register(feature);
+  }
+}
+
+function loadRegistrations() {
+  return fetch(`/registrations/${deviceId}`, {
+    method: 'get',
+    headers: { 'Content-type': 'application/json' },
+  })
+  .then(handleRegistrationsResponse);
+}
+
+// generate a random string (default: 40)
+function makeId(length) {
+  const arr = new Uint8Array((length || 40) / 2);
+  window.crypto.getRandomValues(arr);
+  return [].map.call(arr, n => n.toString(16)).join('');
+}
+//
+// subscribeButton.onclick = function registerAll() {
+//   register('all');
+// };
+//
+// unsubscribeButton.onclick = function unRegisterAll() {
+//   unregister();
+// };
+
+window.onload = () => {
+  if (!navigator.serviceWorker) {
+    console.log('No service workers allowed');
+    return;
+  }
+
+  resetNotifications(toggleNotification);
+
+  localforage.getItem('deviceId')
+  .then(id => {
+    if (id) {
+      deviceId = id;
+      loadRegistrations();
```

Comment:
Yes, there will be some additional work to be sure all usecases are working (unsubscribing from notifications and such)